app/ui-message/client/messageBubble.js

```diff
@@ -0,0 +1,315 @@
+};
+
+Template.messageBubble.helpers({
+	unread() {
+		const { msg, subscription } = this;
+		return subscription?.tunread?.includes(msg._id);
+	},
+	body() {
+		const { msg, settings } = this;
+		return Tracker.nonreactive(() => renderBody(msg, settings));
+	},
+	attachmentsClass() {
+		const { msg } = this;
+
+		if (msg?.attachments?.length) {
+			if (msg.attachments[0]?.image_type) {
+				return 'attachment image';
+			}
+
+			return 'attachment file';
+		}
+
+		if (msg.urls?.length) {
+			return 'url';
+		}
+	},
+	hasAttachments() {
+		const { msg } = this;
+		return msg.attachments?.length;
+	},
+	getFileIcon() {
+		const { msg } = this;
+
+		if (msg?.attachments?.length) {
+			if (msg.attachments[0]?.image_type) {
+				return false;
+			}
+			if (msg.attachments[0]?.audio_type) {
+				return false;
+			}
+			if (msg.attachments[0]?.video_type) {
+				return false;
+			}
+
+			let ext = msg.attachments[0]?.title_link;
+			ext = ext.substring(ext.lastIndexOf('.'));
+
+			if (ext === '.doc') {
+				return 'file-document';
+			}
+			if (ext === '.pdf') {
+				return 'file-pdf';
+			}
+			if (ext === '.key') {
+				return 'file-keynote';
+			}
+			if (ext === '.zip') {
+				return 'files-zip';
+			}
+
+			return 'file-generic';
+		}
+	},
+	own() {
+		const { msg, u = {} } = this;
+		return msg.u && msg.u._id === u._id ? 'own' : 'notOwn';
+	},
+	bigEmoji() {
+		let { msg } = this;
+		msg = renderMessageBody(msg);
+		const classesIndex = msg.indexOf('class="');
+
+		if (classesIndex > -1) {
+			const classes = msg.substring(classesIndex + 7, msg.indexOf('"', classesIndex + 7)).split(' ');
+			if (classes.includes('big')) {
+				return 'bigEmoji';
+			}
+		}
+
+		return '';
+	},
+	timestamp() {
+		const { msg } = this;
+		return +msg.ts;
+	},
+	hasOembed() {
+		const { msg, settings } = this;
+		// there is no URLs, there is no template to show the oembed (oembed package removed) or oembed is not enable
+		if (!(msg.urls && msg.urls.length > 0) || !Template.oembedBaseWidget || !settings.API_Embed) {
+			return false;
+		}
+		// check if oembed is disabled for message's sender
+		if ((settings.API_EmbedDisabledFor || '').split(',').map((username) => username.trim()).includes(msg.u && msg.u.username)) {
+			return false;
+		}
+		return true;
+	},
+	injectMessage(data, { _id, rid }) {
+		data.msg = { _id, rid };
+	},
+	injectIndex(data, index) {
+		data.index = index;
+	},
+	injectSettings(data, settings) {
+		data.settings = settings;
+	},
+	actionContext() {
+		const { msg } = this;
+		return msg.actionContext;
+	},
+	shouldHideBody() {
+		const { msg: { tmid, actionContext }, settings: { showreply }, context } = this;
+		return showreply && tmid && !(actionContext || context);
+	},
+	readReceipt() {
+		if (!settings.get('Message_Read_Receipt_Enabled')) {
+			return;
+		}
+
+		return {
+			readByEveryone: (!this.msg.unread && 'read') || 'color-component-color',
+		};
+	},
+});
+
+const hasTempClass = (node) => node.classList.contains('temp');
```

Comment:
Can we rename it from temp class to actual purpose of this class to make it more clear

Reply:
This function refers to the actual 'temp class' which was here since before UI migration. and is not something specifically added in this PR.
Probably we shouldn't rename the class definition

But we can change the function variable name (but that looks fine considering the classname)